### DISCUSSION TITLE
Ideas

### DISCUSSION THREAD

**Discussion by iamDyeus** [iamDyeus/KnickAI#2] [Ideas]
Give your **Ideas/Suggestions** Below :)

**Comment by vednig** [comment]
Could you provide the exe

**Reply by iamDyeus** [reply]
Here is the **Setup** : https://bit.ly/knick_setup

**Reply by vednig** [reply]
The link does not seem to be working. 
I would suggest you add up exe to the release section.
This way it would be over here without GoogleDrive removing it.

**Reply by iamDyeus** [reply]
I'll put a .zip of source code as well as the setup in the release section soon. its just that I am waiting to finish with the TTS and bugs in it first, so that I can remake a new setup that is more reliable.

For the Time Being You may Use this Link to Download the Setup if you want : https://bit.ly/3Fg1a2U

**Comment by vednig** [comment]
Suggestions:

1. Ability to run in background
2. Open Dictation Option by saying "I want to dictate"
3. Make interaction more natural.(Instead of repeating ask "anything more/else i can do for you?" and hibernate after the user says "no/thank you" etc. and wake on "Knick" - Hot Word Detection)
4. Save User Preferences
5. Respond to different users based on their stored preferences.

**Reply by iamDyeus** [reply]
if you could describe each feature your telling about in brief, then I can make a Different Discussion for each one and start working on them.

**Reply by vednig** [reply]
#### Feature Suggestion:

- Ability to run in background

#### Implementation/Suggested Behavior:

When 
```python 
run_in_background=="Activated"

``` 
saying 

> Hey Knick/Knick Take screenshot

 should take a screen shot and respond with 

> "Screenshot taken and Saved to [Preference Location]"

**Reply by vednig** [reply]
#### Feature Suggestion:

- Open Dictation Option by saying "I want to dictate"

#### Implementation/Suggested Behavior:

When 
```python 
said_text=="I want to dictate"

``` 
saying 

> Hey Knick/Knick Take screenshot

 should 

1. Open dictation panel( Win +H ) via pyautogui 
2. And enter received input on chrome and respond with 

> "You can start dictating now.[Preference Title/Salutation]"

**Reply by vednig** [reply]
#### Feature Suggestion:

- Make interaction more natural.(Instead of repeating ask "anything more/else i can do for you?" and hibernate after the user says "no/thank you" etc. and wake on "Knick" - Hot Word Detection)

#### Implementation/Suggested Behavior:

When 
command completed respond with 

> "Anything Else? [Preference Location]"(for new command)


if responded with 'no'
Go to Hibernation(sleep/not doing or speaking anything except running in background) and Wake on "Wake Word"(Knick)

**Reply by vednig** [reply]
#### Feature Suggestion:

- Save User Preferences

#### Implementation/Suggested Behavior:
Create `preferences.db` and store User Preferences (to be done on initialization)

**Reply by vednig** [reply]
#### Feature Suggestion:

- Respond to different users based on their stored preferences.

#### Implementation/Suggested Behavior:

Analyze voice and use change in voice pitch to create virtual users and store their preferences.

**Reply by iamDyeus** [reply]
Decided a sequence to start working on some of your suggested Features,  

- Ability to Work in Background + Hotward Detection 
- "i want to dictate" feature. (Probably will just press Win+H on command) 

though i think in the second feature, i wont be adding a feature to search for that dictated text in chrome as there is some feature already in the assistant that is similar to it. I'll just try to make that feature a bit better.

As for the Preferences feature, I'll just keep it in [concept](https://github.com/iamDyeus/KnickAI/projects/1#column-17304894) because already have a lot to-do.

**Reply by vednig** [reply]
Great🍟
